mods/linked/moves.js

```diff
@@ -0,0 +1,462 @@
+			return false;
+		},
+	},
+
+	/**
+	 *	Sucker Punch
+	 *	Will miss on two linked Status moves
+	 *
+	 */
+
+	suckerpunch: {
+		inherit: true,
+		onTry: function (source, target) {
+			let action = this.willMove(target);
+			if (!action || action.choice !== 'move') {
+				this.attrLastMove('[still]');
+				this.add('-fail', source);
+				return null;
+			}
+			if (target.volatiles.mustrecharge && target.volatiles.mustrecharge.duration < 2) {
+				// Duration may not be lower than 2 if Sucker Punch is used as a low-priority move
+				// i.e. if Sucker Punch is linked with a negative priority move
+				this.attrLastMove('[still]');
+				this.add('-fail', source);
+				return null;
+			}
+			if (!target.linked && action.move.category === 'Status' && action.move.id !== 'mefirst') {
+				this.attrLastMove('[still]');
+				this.add('-fail', source);
+				return null;
+			}
+			if (target.linked) {
+				for (const linkedMoveid of target.linked) {
+					let linkedMove = this.getMove(linkedMoveid);
+					if (linkedMove.category !== 'Status' || linkedMove.id === 'mefirst') return;
+				}
+				this.attrLastMove('[still]');
+				this.add('-fail', source);
+				return null;
+			}
+		},
+	},
+
+	/**
+	 * Mimic and Sketch
+	 * When any of them is linked, the link will get updated for the new move
+	 * They will copy the last absolute single move used by the foe.
+	 *
+	 **/
+
+	sketch: {
+		inherit: true,
+		onHit: function (target, source) {
+			let disallowedMoves = ['chatter', 'sketch', 'struggle'];
+			let lastMove = target.getLastMoveAbsolute();
+			if (source.transformed || !lastMove || disallowedMoves.includes(lastMove.id) || source.moves.indexOf(lastMove.id) >= 0 || lastMove.isZ) return false;
+			let sketchIndex = source.moves.indexOf('sketch');
+			if (sketchIndex < 0) return false;
+			let move = this.getMove(lastMove.id);
+			let sketchedMove = {
+				move: move.name,
+				id: move.id,
+				pp: move.pp,
+				maxpp: move.pp,
+				target: move.target,
+				disabled: false,
+				used: false,
+			};
+			source.moveSlots[sketchIndex] = sketchedMove;
+			source.baseMoveSlots[sketchIndex] = sketchedMove;
+			this.add('-activate', source, 'move: Sketch', move.name);
+		},
+	},
+	mimic: {
+		inherit: true,
+		onHit: function (target, source) {
+			let disallowedMoves = ['chatter', 'mimic', 'sketch', 'struggle', 'transform'];
+			let lastMove = target.getLastMoveAbsolute();
+			if (source.transformed || !lastMove || disallowedMoves.includes(lastMove.id) || source.moves.indexOf(lastMove.id) >= 0 || lastMove.isZ) return false;
+			let mimicIndex = source.moves.indexOf('mimic');
+			if (mimicIndex < 0) return false;
+			let move = this.getMove(lastMove.id);
+			source.moveSlots[mimicIndex] = {
+				move: move.name,
+				id: move.id,
+				pp: move.pp,
+				maxpp: move.pp,
+				target: move.target,
+				disabled: false,
+				used: false,
+				virtual: true,
+			};
+			this.add('-start', source, 'Mimic', move.name);
+		},
+	},
+
+	/**
+	 * Copycat and Mirror Move
```

Comment:
What happened to Copycat?

Reply:
removed because the code was unnecessary